src/theory/quantifiers/first_order_model.cpp

```diff
@@ -55,12 +56,11 @@ unsigned FirstOrderModel::getNumAssertedQuantifiers() {
 }
 
 Node FirstOrderModel::getAssertedQuantifier( unsigned i, bool ordered ) { 
-  if( !ordered ){
+  if( !ordered || !d_forallRlvComputed ){
     return d_forall_asserts[i]; 
-  }else{
-    Assert(d_forall_rlv_assert.size() == d_forall_asserts.size());
-    return d_forall_rlv_assert[i];
   }
+  Assert(d_forall_rlv_assert.size() == d_forall_asserts.size());
```

Comment:
Why not use this test rather than introduce a new flag?

Reply:
Using the flag makes it explicit that the data is not stale.  Otherwise we are relying on a hidden invariant that this condition is equivalent to the fact that d_forall_rlv_assert is not stale.

Reply:
The actual issue is the invariant being hidden hehe The flag did not help me much, to understand what was going on I still had to look over the code to discover that invariant. Maybe just saying it explicitly above this line would suffice.